designer/client/src/ComponentTypeEdit.tsx

```diff
@@ -55,7 +52,8 @@ export const ComponentTypeEdit: FunctionComponent<Props> = (props) => {
     ? componentTypeEditors[selectedComponent.type]
     : undefined
 
-  const { type, subType } = component ?? {}
+  const defaults = getComponentDefaults(selectedComponent)
```

Comment:
We can't reliably pull `type` or `subType` from the form definition JSON 😣

Like lots of other places, we should refer to the defaults for these properties